### DISCUSSION TITLE
Excel-DNA 1.6.0 release - .NET 6 support

### DISCUSSION THREAD

**Discussion by MarkHone** [Excel-DNA/ExcelDna#628] [Q&A] [answered]
Hi @govert ,

I've just been reading up on the original Excel-DNA 1.6 preview release blog post from February 2022, where "pre-release", and subsequent posts regarding the final 1.6.0 package release, which looks like it happened around October 2022.

There don't seem to be many posts complaining about issues with .NET 6 so I was just wondering what your view was on the state of .NET 6 support in 1.6.0.  Is it still "pre-release" level or would you consider it production ready yet?  Or, if I wanted to deploy something to production, should I wait for the 1.7 release? (I see 1.7.0-rc5 was released just under a month ago).

Thanks in advance,

Mark

**Comment by govert** [comment]
You should just go with the latest 1.7.0 pre-release or release whenever that happens. The only difference will be when I've had time to write down the list of project settings, and maybe fixed a minor bug or two more. Nothing about 1.6.0 is better or more stable than 1.7.0-rc5, and the new version has fixes and improvements.

**Reply by MarkHone** [reply]
Thanks very much for the speedy reply - appreciated!  Would you class the .NET 6 support in 1.7.0-rc5 as production-ready though?  From my side everything seems to work well (Excel-DNA is an impressive piece of work!), but do you know if other Excel-DNA users have successfully deployed to production?  Sorry to labour the point but I just want to conduct a bit of "due diligence" before feeling ready to take the plunge!

**Comment by govert** [comment, marked as answer]
To be sure, using .NET 6 is still more risky than .NET Framework. There are issues with having the right runtime installed, interactions with other add-ins and also reloading is messy / unsupported. But if you need to move on to .NET 6, this is the best we have at the moment.

**Reply by MarkHone** [reply]
Aha! I missed this while I was replying to your previous comment, so apologies for that.  I understand the risks with the runtime and have read about the reloading scenarios (I imagine not helped by how Excel itself works in this regard).  The risks around interactions with other Add-ins is a relevant point that could be explored further on my side.  Thanks for your answers!
